SpatialGDK/Source/SpatialGDK/Public/SpatialView/EntityView.h

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialView/ComponentData.h"
+#include "SpatialCommonTypes.h"
+
+#include "Containers/Array.h"
+#include "Containers/Map.h"
+#include <improbable/c_worker.h>
+
+namespace SpatialGDK
+{
+
+struct EntityViewElement
+{
+	TArray<ComponentData> Components;
```

Comment:
I think the number of components on a specific entity is probably still small enough that it's easier to iterate although we can revisit this.
Especially as if we need stable pointers then we should revisit this anyway with both things in mind.

Reply:
Also there are times where we iterate through all the data. It's mostly an array as a default - I haven't profiled or thought about this enough but the data sets aren't huge - container.

Reply:
fair enough, can leave this then

Reply:
As I understand this is intended to replace SpatialStaticComponentView, being able to access individual components seems like the most common use-case.

Reply:
It's definitely possible. However it might also end up being getting all relevant components on a newly complete entity that dominates view access (outside the view delta). It could also be the update on send part but that's likely to end up changing to become more entity-centric too. It's a hard one to guess and likely requires a profiling to answer.

Couple of things to consider:

- We don't know if we need stable pointers or sorted data yet. What exact combination we need likely changes what container is best.

- I expect closer to 10 components per entity than 100. In which case I'd expect iterating over an array to find an individual component to be faster or essentially identical to finding it in a hash map. For individual component access this wouldn't be the case if we had a map from entity-component-ID to data but the need to iterate over entities means that isn't really practical here. If it's a lot more than 10 then that would change if it's also unsorted. That said I haven't benchmarked this again TMap so it's hard to guess.

Reply:
Leaving this as is for now for the reasons above and because it costs us nothing to reopen this discussion at any point. It's just an implementation detail.